language/tools/move-cli/src/main.rs

```diff
@@ -107,69 +116,64 @@ enum Command {
 fn maybe_create_dir(dir_name: &str) -> Result<&Path> {
     let dir = Path::new(dir_name);
     if !dir.exists() {
-        fs::create_dir(dir)?
+        fs::create_dir_all(dir)?
     }
     Ok(dir)
 }
 
-/// Compile the user modules in `move_src` and the script in `script_file`
-fn compile(
-    args: &Move,
-    script_file: &Option<String>,
-) -> Result<(OnDiskStateView, Option<CompiledScript>)> {
-    let move_data = maybe_create_dir(&args.move_data)?;
+/// Generate interface files for published files
+fn generate_interface_files(args: &Move) -> Result<()> {
+    move_lang::generate_interface_files(
+        &[args.move_data.clone()],
+        Some(args.move_data.clone()),
+        false,
+    )?;
+    Ok(())
+}
 
-    // TODO: prevent cyclic dep?
-    let mut user_move_src_files = if Path::new(&args.move_src).exists() {
-        move_lang::find_move_filenames(&[args.move_src.clone()], true)?
-    } else {
-        vec![]
-    };
+fn interface_files_dir(args: &Move) -> Result<String> {
+    let dir = format!("{}/{}", &args.move_data, MOVE_COMPILED_INTERFACES_DIR);
```

Comment:
Nit: use `Path` + `push` to avoid depending on `/` as a directory delimiter.

Reply:
Very fancy. Will be doing another PR in move source lang land to clean that up